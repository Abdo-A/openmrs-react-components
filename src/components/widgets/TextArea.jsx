import React from 'react';
import PropTypes from 'prop-types';
import {  FormControl } from 'react-bootstrap';
import '../../../assets/css/widgets.css';

const TextArea = ({
  input,
  mode,
  placeholder,
  displayValue,
}) => {

  const edit = (
    <FormControl
      {...input}
      componentClass="textarea"
      placeholder={placeholder}
    />
  );

  const view = (
    <span
      className="textarea-input-view"
    >
      {displayValue}
    </span>
  );

  return (
    <div>
      {!mode || mode === 'edit' ? edit : view}
    </div>
  );

};

TextArea.propTypes = {
  input: PropTypes.object,
  meta: PropTypes.object,
  placeholder: PropTypes.string,
  type: PropTypes.string,
};

export default TextArea;



