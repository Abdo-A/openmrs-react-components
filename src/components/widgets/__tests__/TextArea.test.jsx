import React from 'react';
import renderer from 'react-test-renderer';
import TextArea from '../TextArea';

let props;
let mountedComponent;

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = renderer.create(<TextArea {...props} />);
  }
  return mountedComponent;
};

describe('Component: TextArea', () => {
  beforeEach(() => {
    props = {
      input: {
        name: 'mockname'
      },
      placeholder: 'mockplaceholder',
      value: 'mockvalue',
      displayValue: 'displayValue',
    };
    mountedComponent = undefined;
  });

  it('renders properly', () => {
    const component = getComponent();
    expect(component.toJSON()).toMatchSnapshot();
  });
});
