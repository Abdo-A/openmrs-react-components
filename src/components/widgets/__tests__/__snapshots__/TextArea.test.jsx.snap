// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TextArea renders properly 1`] = `
<div>
  <textarea
    className="form-control"
    id={undefined}
    name="mockname"
    placeholder="mockplaceholder"
    type={undefined}
  />
</div>
`;
